pkg/sdn/registry/clusternetwork/etcd/etcd.go

```diff
@@ -0,0 +1,49 @@
+package etcd
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+
+	"github.com/openshift/origin/pkg/sdn/api"
+	"github.com/openshift/origin/pkg/sdn/registry/clusternetwork"
+)
+
+// rest implements a RESTStorage for sdn against etcd
+type REST struct {
+	etcdgeneric.Etcd
+}
+
+const EtcdPrefix = "/registry/sdn/networks"
+
+// NewREST returns a RESTStorage object that will work against subnets
+func NewREST(h tools.EtcdHelper) *REST {
+	store := &etcdgeneric.Etcd{
+		NewFunc:     func() runtime.Object { return &api.ClusterNetwork{} },
+		NewListFunc: func() runtime.Object { return &api.ClusterNetwork{} },
+		KeyRootFunc: func(ctx kapi.Context) string {
+			return EtcdPrefix
+		},
+		KeyFunc: func(ctx kapi.Context, name string) (string, error) {
+			return (EtcdPrefix + "/" + name), nil
+		},
+		ObjectNameFunc: func(obj runtime.Object) (string, error) {
+			return obj.(*api.ClusterNetwork).Name, nil
+		},
+		PredicateFunc: func(label labels.Selector, field fields.Selector) generic.Matcher {
+			return clusternetwork.MatchClusterNetwork(label, field)
+		},
+		EndpointName: "sdn",
```

Comment:
We interchange terminology between cluster network and sdn throughout this PR.  It looks like end-users deal with ClusterNetwork resources, so should this be "clusternetwork"?  I know it just affects error messages, but noticed this intermix of terminology late in my PR review.  It's a nit.